common/src/main/java/io/pravega/common/concurrent/FutureHelpers.java

```diff
@@ -183,12 +183,11 @@
      * @param <ResultT>            Type of the result.
      * @param <ExceptionT>         Type of the Exception.
      * @throws ExceptionT       If thrown by the future.
-     * @throws TimeoutException If the timeout expired prior to the future completing.
-     * @return The result of calling future.get().
+     * @return The result of calling future.get() or null if the timeout expired prior to the future completing.
      */
     @SneakyThrows(InterruptedException.class)
     public static <ResultT, ExceptionT extends Exception> ResultT getAndHandleExceptions(Future<ResultT> future,
-                                                                                         Function<Throwable, ExceptionT> exceptionConstructor, long timeoutMillis) throws TimeoutException, ExceptionT {
```

Comment:
The only caller of this version of the method is in SegmentInputStreamImpl.readEventData()